### PR TITLE
Fix inconsistent input field size in close account page message field

### PR DESCRIPTION
**Pull request by reinaldosg**
### Fixed Issues

$ https://github.com/Expensify/App/issues/20904
PROPOSAL: [#20904(COMMENT)](https://github.com/Expensify/App/issues/20904#issuecomment-1594718234)

### Tests

1. Open Setting
2. Open Security
3. Open Close Account
4. See if the message text field is only one line and auto grow
5. Close Setting
6. Open Any Workspace room
7. Open room details
8. Open setting then welcome message
9. See if the welcome message text field is only one line and auto grow

- [X] Verify that no errors appear in the JS console

### Offline tests

N/A

### QA Steps

Same as test steps
- [X] Verify that no errors appear in the JS console

### PR Author Checklist

- [X] I linked the correct issue in the `### Fixed Issues` section above
- [X] I wrote clear testing steps that cover the changes made in this PR
    - [X] I added steps for local testing in the `Tests` section
    - [X] I added steps for the expected offline behavior in the `Offline steps` section
    - [X] I added steps for Staging and/or Production testing in the `QA steps` section
    - [X] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [X] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I ran the tests on **all platforms** & verified they passed on:
    - [X] Android / native
    - [X] Android / Chrome
    - [X] iOS / native
    - [X] iOS / Safari
    - [X] MacOS / Chrome / Safari
    - [X] MacOS / Desktop
- [X] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [X] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [X] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] I verified that if a function's arguments changed that all usages have also been updated correctly
- [X] If a new component is created I verified that:
    - [X] A similar component doesn't exist in the codebase
    - [X] All props are defined accurately and each prop has a `/** comment above it */`
    - [X] The file is named correctly
    - [X] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [X] The only data being stored in the state is data necessary for rendering and nothing else
    - [X] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [X] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [X] All JSX used for rendering exists in the render method
    - [X] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [X] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/17044978/cacabedc-05a4-4187-a0f7-6211b155d4bd

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/assets/17044978/b451f815-6dd6-4569-ba86-eeceea957dac

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/17044978/f4bd5266-d82b-425f-af15-4263cba03b14

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/17044978/10cd855d-c172-4b39-8b96-ee646f7f9718

</details>

<details>
<summary>iOS</summary>

https://github.com/Expensify/App/assets/17044978/a3ad0d4e-2064-4e94-832e-d8685b77904b

</details>

<details>
<summary>Android</summary>

https://github.com/Expensify/App/assets/17044978/31e41993-849e-440a-b358-328721de5336

</details>

### CODE REVIEW

**Comment by srikarparsi** [comment]
hey @NikkiWines, sorry for the assign, I think it might've been linked differently

**Comment by reinaldosg** [comment]
recheck

**Review by venture1981 — COMMENTED** [review]
@reinaldosg 
seems like you changed form component.
btw, you updated the docs in [FORMS.md] accordingly?

**Comment by reinaldosg** [comment]
@venture1981 I think my changes aren't worth mentioning on the docs since it doesn't affect how to implement the Form component. But let me know if you have a different opinion 🙏

**Comment by reinaldosg** [comment]
recheck

**Comment by reinaldosebastian** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by reinaldosebastian** [comment]
recheck

File: src/components/TextInput/BaseTextInput.js [hunk 1]
```diff
@@ -224,6 +224,8 @@ class BaseTextInput extends Component {
             {},
         );
         const isMultiline = this.props.multiline || this.props.autoGrowHeight;
+        //if AutoGrowWeight is true, placeholder needs to be filled with any string.
+        const hiddenTextPlaceholder = this.props.placeholder ? this.props.placeholder : this.props.autoGrowHeight ? ' ' : '';
```
[the remark this answers is not in the document]
Comment (reinaldosg) [reply src/components/TextInput/BaseTextInput.js:228]: @thesahindia Actually this is kind of weird behavior. In the Desktop app, the height isn't set right away if there's no text set on the hidden Text component. (see attached video)

https://github.com/Expensify/App/assets/17044978/a3e8ae92-fe78-46a5-a181-c5b54dac2b40

**Comment by shawnborton** [comment]
> should we keep the max height same as invite message field?

I think it makes sense to keep them consistent, yeah. It should just be a global component that behaves the same everywhere as much as we can.

File: src/components/TextInput/BaseTextInput.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (reinaldosg) [reply src/components/TextInput/BaseTextInput.js]: @thesahindia Oh didn't know that! I pushed the revert, please take a look again when you have time 🙏

**Review by thesahindia — APPROVED** [review]
Tests well!

NOTE: In mWeb and desktop the field height will be 115 px from the start and the height will be auto adjusted on entering some text.

cc: @srikarparsi

**Review by thesahindia — CHANGES_REQUESTED** [review]
@reinaldosg, it seems like the commits aren't signed.

**Comment by reinaldosg** [comment]
@thesahindia Hi! sorry for inconvenience, can I force push the commit so it changes the old unverified commit? or is there anything else to do ?
I tried several things but seems I always added a commit rather than verifying my previous commit. My git configuration is a little bit messed up since I use two accounts on my computer😅

**Comment by reinaldosg** [comment]
@thesahindia Hi sorry for the inconvenience. I found that for signing the existing commit, I need to force push the commit. Is that okay or do I have another way to sign the commit?

**Comment by reinaldosg** [comment]
@thesahindia thanks for the confirmation! Done signing the commit!

**Review by thesahindia — APPROVED** [review]
Thanks @reinaldosg! It looks good now.

**Review by srikarparsi — APPROVED** [review]
looks good to me!

**Comment by srikarparsi** [comment]
Hi @reinaldosg! Tests are failing but I believe merging main into your branch should fix it. Could you try that whenever you get a chance?

**Comment by reinaldosg** [comment]
@srikarparsi done!

**Comment by srikarparsi** [comment]
Thanks @reinaldosebastian and @thesahindia!

**Pull request merged**